### ISSUE TITLE
Credit Card form not so accessiblity compatible for the visually impaired

### ISSUE THREAD

**Issue by imodouglas** [woocommerce/woocommerce-square#41]
### Describe the bug

A merchant in [this forum thread](https://wordpress.org/support/topic/accessibility-concern-potential-issue-for-visually-impaired-users/#post-17213329) raised a complaint about the card input form not being accessibility compatible for the visually impaired users. Also, the MM/YY placeholder is read for that section and can be confusing for the visually impaired.

### To reproduce
1. Setup square on a test site.
2. Turn on the voiceover on your PC.
3. Try to checkout as a visually impaired person.

**Screenshots**

https://d.pr/f/cu1RZa

https://github.com/woocommerce/woocommerce-square/assets/2818133/d9393a2a-0ac1-4314-ab56-90728dce9c8e

### Expected behavior

It should be descriptive enough for the visually impaired person.
